client/my-sites/sidebar-unified/use-site-menu-items.js

```diff
@@ -46,6 +47,13 @@ const useSiteMenuItems = () => {
 		canCurrentUser( state, selectedSiteId, 'edit_theme_options' )
 	);
 
+	/**
+	 * When no site domain is provided, lets show only menu items that support all sites screens.
+	 */
+	if ( ! siteDomain ) {
```

Comment:
Should we also bale out of the `useEffect`s above when there is no site domain as these calls will not be required.

Reply:
I had already tested that we don't call the api with a `null` Domain and it seems it is safeguarded. But you are right! I have addressed it here https://github.com/Automattic/wp-calypso/pull/49026/files together with the "organising all the static data files".